test/generator.cc

```diff
@@ -254,10 +254,9 @@ TEST(Generator, FailStream) {
             functions.done.Call();
           })
           .fail([&](auto& k, auto&& error) {
-            // No need to specify 'raises' because type of error is
-            // 'std::exception_ptr', that just propagates.
+            // Will not be invoked since we do not specify any of
+            // 'Catches' or 'Raises'.
             functions.fail.Call();
```

Comment:
If this won't be invoked then let's put in a `FAIL()` here instead! And I think this is just because we don't specify `Catches` right? Doesn't matter if we specify `Raises`.
```suggestion
            FAIL() << "Should not get called because we did not specify any 'Catches'";
```